internal/locate/region_request.go

```diff
@@ -127,7 +127,11 @@ func (s *RegionRequestSender) String() string {
 
 // RegionRequestRuntimeStats records the runtime stats of send region requests.
 type RegionRequestRuntimeStats struct {
-	RPCStats map[tikvrpc.CmdType]*RPCRuntimeStats
+	// FirstRPCStats is the stats of first kinds of rpc request, since in most cases, only one kind of rpc request is sent at a time,
+	// this is to avoid allocating map memory.
+	FirstRPCStats RPCRuntimeStats
+	// OtherRPCStatsMap uses to record another types of RPC requests.
+	OtherRPCStatsMap map[tikvrpc.CmdType]*RPCRuntimeStats
```

Comment:
Just a suggestion, since one RegionRequestSender typically only send one (or a few) kind(s) of RPC, I guess we can just use a slice here to make the code clean and simple.

Reply:
great, done.